httpx/models.py

```diff
@@ -291,7 +292,16 @@ def __init__(self, *args: QueryParamTypes, **kwargs: typing.Any) -> None:
         elif isinstance(value, list):
             items = value  # type: ignore
         else:
-            items = value.items()  # type: ignore
+            # A dict-like object whose values may be lists.
+            items = [
+                (k, u)  # type: ignore
+                for k, v in value.items()
+                for u in (
+                    v  # type: ignore
+                    if hasattr(v, "__iter__") and not isinstance(v, (str, bytes))
+                    else [v]  # type: ignore
+                )
+            ]
```

Comment:
I think double comprehensions are a bit incomprehensible. Any other tack we could take here maybe?